File: frontend/src/app/model/purchase-request.ts
import { Address } from "./address";
import { Client } from "./client";
import { Item } from "./item";
import { RequestOrder } from "./request-order";

export class PurchaseRequest {



    client: Client;
    fromAddress: Address;
    toAddress: Address;
    requestOrder: RequestOrder;
    items: Item[];

    constructor(client: Client, fromAddress: Address, toAddress: Address, requestOrder: RequestOrder, items: Item[]) {
        this.client = client;
        this.fromAddress = fromAddress;
        this.toAddress = toAddress;
        this.requestOrder = requestOrder;
        this.items = items;
    }
}

File: backend/Restaurant-Project/src/main/java/com/morgan/restaurant/dto/PurchaseRequest.java
package com.morgan.restaurant.dto;

import com.morgan.restaurant.model.Address;
import com.morgan.restaurant.model.Client;
import com.morgan.restaurant.model.Item;
import com.morgan.restaurant.model.RequestOrder;
import lombok.Data;

import java.util.Set;

@Data
public class PurchaseRequest {

    private Client client;

    private RequestOrder requestOrder;

    private Set<Item> items;

    private Address toAddress;

    private Address fromAddress;
}
